pkg/core/tekton/tekton.go

```diff
@@ -0,0 +1,405 @@
+package tekton
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"knative.dev/pkg/apis"
+
+	"github.com/fuseml/fuseml-core/gen/codeset"
+	"github.com/fuseml/fuseml-core/gen/workflow"
+	"github.com/fuseml/fuseml-core/pkg/core/tekton/builder"
+)
+
+// WorkflowBackend implements the FuseML WorkflowBackend interface for tekton
+type WorkflowBackend struct {
+	namespace     string
+	tektonClients *clients
+}
+
+// NewWorkflowBackend initializes Tekton backend
+func NewWorkflowBackend(namespace string) *WorkflowBackend {
+	return &WorkflowBackend{namespace, newClients(namespace)}
+}
+
+// CreatePipeline receives a FuseML workflow and creates a Tekton pipeline from it
+func (t *WorkflowBackend) CreatePipeline(ctx context.Context, logger *log.Logger, workflow *workflow.Workflow) error {
```

Comment:
All these are methods for WorkflowBackend struct ... why is the variable (= backend instance) called `t` ?

Reply:
Indeed `t` makes no sense, I have changed the type name and did not update it here.